code/go/0chain.net/chaincore/block/entity.go

```diff
@@ -1008,6 +1008,11 @@ func (b *Block) ComputeStateLocal(ctx context.Context, c Chainer) error {
 		}
 	}
 
+	err := emitBlockEvent(b)
+	if err != nil {
+		return err
```

Comment:
Is there any function which makes sure that the block object has all the fields populated which we need, or is it by design that block will have each & every field populated? Should we keep this check for a fail safe, what repurcussions can we have if we return an error here?

Reply:
I think it is correct to log the error rather than throwing back the error as it can stop other events and subsequent process just because of events. Also, even if there are some fields that are not present, then they would just be `null` after marshal.